DOC-3356: New feature documentation for `tinymceai`. (#3976)
tinymce/tinymce-docs: modules/ROOT/pages/tinymceai-review-plugin.adoc

= TinyMCE AI Review
:navtitle: Review
:pluginname: TinyMCE AI
:plugincode: tinymceai
:description: AI Review feature for TinyMCE AI plugin
:description_short: AI Review feature
:keywords: AI, review, reviews, tinymceai
:page-role: -toc

> The Review feature provides AI-powered quality assurance for content by running checks for grammar, style, tone, and more. It introduces an intuitive interface for reviewing and managing AI-suggested edits directly within the document, ensuring content meets professional standards with minimal manual effort.

Reviewer: ```suggestion
The Review feature provides AI-powered quality assurance for content by running checks for grammar, style, tone, and more. The plugin introduces an intuitive interface for reviewing and managing AI-suggested edits directly within the document, ensuring content meets professional standards with minimal manual effort. The API offers the same functionality with flexible integration.
```
just mentioning the API here

Reviewer: ...though maybe that then clashes with the next sentence a bit?

The Review feature is available as a plugin UI and through the Reviews API. This page covers both:

* Plugin sections describe user-facing features and configuration (xref:tinymceai.adoc#options-review[Review configuration]).
* API sections cover REST API access (starting at xref:tinymceai-review-plugin.adoc#reviews-api[Reviews API]).

[[demo]]
== Demo

liveDemo::tinymceai-review[]

> The Review feature analyzes content and provides specific recommendations for grammar, style, clarity, and tone improvements. Unlike Actions that transform content directly, Reviews analyze content and provide suggestions without automatically changing the content.

Reviewer: not sure how to reword this, but Actions don't change the content immediately - you still have to accept or reject changes. some of the operations _run_ immediately (e.g. the chat ones) but that's different. to me, the main difference is that Actions work on a section of the document (sometimes that can be the whole document, but still) and can do a wide variety of operations (trigger chat, run a prompt, translate, etc.) whereas Reviews are always the whole document and are specifically designed for content operations where you review suggestions

Reviewer: also, this paragraph gets a little lost between the big blocks of the example before it and screenshot after it?

Author: This was a challenging one but i think we got it in the end. see [this commit](https://github.com/tinymce/tinymce-docs/pull/3976/changes/8135a5e49ecbd389e6aed301c54fb5ca475aeb21)

image::tinymceai/tinymce-ai-review-sidebar-command-categories-proofread-clarity.png[{pluginname} Review feature showing AI-powered quality assurance and suggested edits,width=80%]

[[integration]]
== Integration

> To start using the Review feature, first load the {pluginname} plugin in the editor configuration. The Review Mode button image:icons-premium/ai-review.svg[Review icon,24px] will appear in the AI user interface. See xref:tinymceai.adoc[Plugin Reference] for installation and enabling AI features.

Reviewer: ```suggestion
To start using the Review feature,  add `+tinymceai+` to the plugins option. This will automatically add the image:icons-premium/ai-review.svg[Review icon,24px] Review toolbar button and menu item to the default {productname} toolbar and menu.
 
See xref:tinymceai.adoc[Plugin Reference] for more information regarding installation and enabling AI features.
```
to match my suggestion on Chat

Reviewer: I should add, this just feels clearer? the wording to start using

> After picking one of the available commands in the Review Mode tab, the AI will analyze the document and propose a series of suggestions:

Reviewer: ```suggestion
After picking one of the available reviews in the Review Mode tab, the AI will analyze the document and propose a series of suggestions:
```

image::tinymceai/tinymce-ai-review-improve-readability-suggestion-cards-and-underlines.png[{pluginname} Review sidebar with suggestions,width=80%]

While in the Review Mode, the editor remains read-only and allows browsing suggestions. Suggestions can be clicked in the sidebar or selected in the editor content (underlined):

> image::tinymceai/tinymce-ai-review-mode-suggestions-underlined-in-editor-content.png[{pluginname} Review suggestion in content,width=80%]

Reviewer: Is this screenshot in the right place? The previous screenshot feels more fitting for this paragraph?

Author: Nice pickup, reversed the order of the images as suggested: see [commit](https://github.com/tinymce/tinymce-docs/pull/3976/commits/e16751f855919886f54f9b826534f15e48859860)

Review suggestions can be accepted or dismissed by clicking the corresponding buttons. All suggestions can also be accepted using the "Accept all" button at the top of the user interface. Changes can be previewed in the same way as xref:tinymceai-chat.adoc#previewing-changes[Chat suggestions]. Changes that were accepted or dismissed become greyed out in the interface. The review can also be abandoned by clicking the "Exit review" button.

> Once reviewing is complete and all changes are accepted or rejected, click "Finish review" (the button state changes automatically) to return to the normal operation of the editor, where typing is possible.

Reviewer: ```suggestion
Review suggestions can be individually skipped or applied by clicking the corresponding buttons. Changes that were accepted or dismissed become greyed out in the interface. The "Complete" button in the top of the sidebar enables actions such as:

- Skipping or applying the remaining suggestions in bulk
- Completing the review with the currently applied suggestions (and skipping the rest)
- Aborting the review and exiting (with all applied changes reverted)
```

Reviewer: content was wrong. we ended up implementing it differently

[[review-commands]]
== Review commands

Built-in system reviews are listed below. The **Editor command** column is the label in the Review sidebar when that review is available in the editor. The **API review** column is the system review name used in the https://tinymceai.api.tiny.cloud/docs#tag/Reviews[Reviews API].

[cols="1,1,3,^,^",options="header"]
|===

> |Editor command |API review |Description |Editor UI |API

Reviewer: almost missed this - bit wary of using "commands" here because term overload. we _will_ be adding commands for triggering reviews later too, which could make it even more confusing at that point. operations, maybe? or just "reviews" if we can get away with it?

Author: Good catch. Renamed the section to Built-in reviews, updated the anchor to built-in-reviews, and changed the first column header from “Editor command” to Editor so it doesn’t collide with editor/API “commands” later. Surrounding copy now says “built-in reviews” / “listed in the sidebar” where it used to say “commands” for that table. For more details see: [this commit](https://github.com/tinymce/tinymce-docs/pull/3976/commits/2c1ecc983d7f986abfe7c3c19738a5369cbe92b0)

|**Proofread** |Correctness |Check grammar, spelling, and punctuation; the correctness API review also addresses factual errors. |✓ |✓

|**Improve clarity** |Clarity |Improve logical structure, precision, sentence structure, word choice, and flow. |✓ |✓

|**Improve readability** |Readability |Improve readability through sentence structure, paragraph structure, transitions, word choice, and reading level. |✓ |✓

|**Adjust length** |Length Optimization |Shorten or lengthen the text while preserving key information. The editor offers _Longer_ and _Shorter_ options. |✓ |✓

|**Adjust tone and style** |Tone Adjustment |Adjust tone and style. The editor includes _Casual_, _Direct_, _Friendly_, _Confident_, and _Professional_ options. |✓ |✓

| |Translation |Translate between languages with cultural adaptation. | |✓
|===

[NOTE]
====

> * A checkmark in the Editor UI column means that review type can appear in the Review sidebar when its command is listed in `tinymceai_reviews`.

Reviewer: ...should we list the names people need to use with that option here, or just link them to the plugin page section for that option? I vote the latter

Reviewer: just noticed the link is just below, so maybe ignore this? up to you

Author: Updated in [this commit](https://github.com/tinymce/tinymce-docs/pull/3976/changes/ed724ab34b648cc1e678055ef4b242645de3c671)

* Custom reviews (analysis defined with a prompt) are available through the xref:tinymceai-review-plugin.adoc#reviews-custom-reviews[Custom reviews (API)]. _Note: The same capability in the Review sidebar is coming soon._

> * Translation is API-only (no Review sidebar command); see the Translation row in the table above.

Reviewer: ```suggestion
* Translation Review is API-only (no Review sidebar command); see the Translation row in the table above. For an editor translation operation, see the Translate Quick Action.
```
needs a link, idk how sorry

Author: Updated in [this commit](https://github.com/tinymce/tinymce-docs/pull/3976/commits/62d02963f126afb601220448a4e2ff2c1c8987c5)

====

[[review-configuration]]
== Configuration

Which review commands appear in the Review sidebar can be configured using the `tinymceai_reviews` option. See xref:tinymceai.adoc#tinymceai_reviews[`tinymceai_reviews`] in the Plugin Reference for details.

[[reviews-api]]
== Reviews API

> The Review plugin feature is built on top of the Reviews API, which provides REST API access to review functionality. Reviews provide comprehensive content analysis and improvement suggestions. Unlike Actions that transform content directly, Reviews analyze content and provide specific recommendations for grammar, style, clarity, and tone improvements.

Reviewer: same problem with the comparison to Actions as above

Author: Thanks for calling that out. The Reviews API section no longer contrasts Reviews with Actions using overlapping examples (e.g. grammar/tone). It now describes Reviews as a whole-document workflow in Review mode, and Quick Actions as shortcuts to discrete AI operations on a selection or range. see [update](https://github.com/tinymce/tinymce-docs/pull/3976/commits/2c1ecc983d7f986abfe7c3c19738a5369cbe92b0)

> **When to use Reviews vs Actions:** Use Reviews when analyzing entire documents for quality improvements and getting suggestions without automatically changing the content. Use Actions when transforming specific text content (fix grammar, translate, adjust tone).

Reviewer: the examples under actions aren't great, because they also mostly apply to reviews. we need to reframe this in terms of Quick Actions being shortcuts to a variety of AI features, whereas Reviews is a whole-document workflow kind of thing

Author: Agreed—the old “fix grammar / adjust tone” style split blurred the two. Reframed: Reviews = end-to-end whole-document workflow with the Review sidebar and accept/reject. Quick Actions = shortcuts from the Quick Actions menu to individual AI operations (preview, translation, chat, etc.), not that same workflow. The intro and Related features line were aligned with that wording too. see [this commit](https://github.com/tinymce/tinymce-docs/pull/3976/commits/2c1ecc983d7f986abfe7c3c19738a5369cbe92b0) for more details is required.

Reviews use streaming output with Server-Sent Events for real-time feedback as suggestions are generated. Each review type is optimized for specific improvement tasks, providing consistent, high-quality analysis of text structure, style, and quality. Reviews provide specific, actionable recommendations for content improvement.

[[reviews-system-reviews-api]]
=== System Reviews

> Built-in system reviews including API names, descriptions, and whether each type is available in the Review sidebar or only through the API—are summarized in xref:tinymceai-review-plugin.adoc#review-commands[Review commands].

Reviewer: ```suggestion
System reviews are predefined review operations for common use cases, which each use the AI agent most suitable for their functionality. The API includes one endpoint for running system reviews, which is passed the review name. 

Built-in system reviews including API names, descriptions, and whether each type is available in the Review sidebar or only through the API—are summarized in xref:tinymceai-review-plugin.adoc#review-commands[Review commands].
```

For endpoint details, request and response schemas, authentication, and streaming behavior, see https://tinymceai.api.tiny.cloud/docs#tag/Reviews[Reviews API].

[[reviews-custom-reviews]]
=== Custom reviews (API)

> Custom reviews use a unified Reviews API endpoint: analysis behavior is defined with a prompt parameter instead of the fixed system review identifiers listed under xref:tinymceai-review-plugin.adoc#review-commands[Review commands]. See https://tinymceai.api.tiny.cloud/docs#tag/Reviews[Reviews API] for the endpoint, parameters, and examples.

Reviewer: ```suggestion
Integrators can also define custom reviews using the Custom Review API endpoint, which returns a output in a similar manner to system reviews but which takes a custom prompt to define the analysis behavior. Custom reviews can be as simple as checking for specific words or phrases, or as complex as applying a company style guide that includes business logic, specific terminology, and more.

 See https://tinymceai.api.tiny.cloud/docs#tag/Reviews[Reviews API] for the endpoint, parameters, and examples.
```

After picking one of the available commands in the Review Mode tab, AI will analyze the document and propose a series of suggestions:

image::https://placehold.net/default.png[TinyMCE AI Review sidebar with suggestions]

While in the Review Mode, the editor remains read-only and allows you to browse suggestions. You can either click suggestions in the sidebar or select them in the editor content (underlined):

image::https://placehold.net/default.png[TinyMCE AI Review suggestion in content]

You can accept or dismiss review suggestions by clicking the corresponding buttons. You can also accept all suggestions by using the "Accept all" button in the top of the user interface and xref:tinymceai-chat.adoc#previewing-changes[preview changes similar to chat suggestions]. Changes that were accepted or dismissed become greyed out in the interface. You can also abandon the review by hitting the "Exit review" button.

Once you are done reviewing your document and all changes are accepted or rejected, click "Finish review" (the button state changes automatically) to return to the normal operation of the editor, where typing is possible.

[[review-commands]]
== Review commands

> The feature comes with several review commands:

Reviewer: We currently don't support custom commands or translation. That may change, but.

Author: @metricjs 

Updated:
- Removed "Custom command" and "Translate" rows from the commands table
- Added NOTE clarifying that custom commands and translation are not currently supported (may change in future updates)
- [x] Do we want to have this NOTE, or shall we remove? @metricjs 
- Removed entire "Customizing translation commands" section including `tinymceai_reviews_translations` configuration documentation

Updated in https://github.com/tinymce/tinymce-docs/pull/3976/commits/7bb949474064a56214a6d0743076fb10e7e485d9
